compiler/il/OMRResolvedMethodSymbol.cpp

```diff
@@ -2387,6 +2388,32 @@ OMR::ResolvedMethodSymbol::setUsesSinglePrecisionMode(bool b)
    _methodFlags.set(SinglePrecisionMode, b);
    }
 
+bool
+OMR::ResolvedMethodSymbol::isParmVariant(TR::ParameterSymbol * parmSymbol)
+   {
+   if (_variantParms == NULL)
+      self()->detectVariantParms();
+   TR_ASSERT_FATAL(parmSymbol->getOrdinal() < self()->getNumParameterSlots(),
+                  "Parm %d (%p) cannot be owned by current method that only has %d parms", parmSymbol->getOrdinal(), parmSymbol, self()->getNumParameterSlots());
+   TR_ASSERT_FATAL(self()->getParmSymRef(parmSymbol->getSlot())->getSymbol()->getParmSymbol() == parmSymbol,
+                  "Parm %p is not owned by current method %s", parmSymbol, self()->signature(self()->comp()->trMemory()));
+   return _variantParms->get(parmSymbol->getOrdinal());
+   }
+
+void
+OMR::ResolvedMethodSymbol::detectVariantParms()
+   {
+   TR_ASSERT_FATAL(self()->getFirstTreeTop() != NULL && self()->getLastTreeTop() != NULL, "Can only detect variant parms for methods with Trees.");
+   if (_variantParms == NULL)
+      _variantParms = new (self()->comp()->trHeapMemory()) TR_BitVector(self()->getNumParameterSlots(), self()->comp()->trMemory(), stackAlloc, notGrowable);
```

Comment:
`_variantParms` is a bit vector for parm slot, but the code is using it as a bit vector for parm. It doesn't matter what we want to use it for, but how we use it should be consistent

Reply:
Fixed in force push.

Reply:
Actually, I think we have to use bit vector for parameters. If we want to detect variant slot, we need to check store to every parm and every auto, which is too conservative for the purpose of this PR, because a slot can be reused for different data type. For example a slot originally holding address type can be reused to hold int32. The slot is variant but the parm is not, because the two uses don't share the same symbol due to different types, we create an auto symbol for the second use, so what the parm symbol refers to remain unchanged. A parm is variant when there is a store to it with the same type, which is what your code is doing.